envoy/stream_info/filter_state.h

```diff
@@ -19,6 +19,32 @@ class FilterState;
 
 using FilterStateSharedPtr = std::shared_ptr<FilterState>;
 
+// Objects stored in the filter state can optionally be shared between the
+// upstream and downstream filter state. Note that sharing happens at the
+// connection level and in some cases may significantly reduce performance by
+// preventing pooling of multiple downstream requests to a single
+// upstream connection.
```

Comment:
Thanks I definitely appreciate this!

So IIRC (and I have about 30% confidence that I do) the values None and SharedWithUpstreamConnectionOnce allow for connection-pooling where multiple clients can use the same backend connection? Or is it just one of those?